contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,564 @@
+Since rkt doesn't use the CRI, the config values would need to be passed in the legacy method.
+
+The runtime option would be placed in the DeviceInfo as such:
+devices = append(devices, kubecontainer.DeviceInfo{PathOnHost: path, PathInContainer: path, Permissions: "mrw"}) for RWO
+devices = append(devices, kubecontainer.DeviceInfo{PathOnHost: path, PathInContainer: path, Permissions: "mr"}) for ROX
+
+Today, this is defaulted always to mrw, thus it would need to be updated with what is passed in.
+
+# Implementation Plan, Features & Milesones
+
+Phase 1: v1.8
+Feature: Pre-provisioned PVs to precreated devices 
+
+               Milestone 1: API changes
+
+               Milestone 2: Restricted Access 
+               
+               Milestone 3: Scrubbing device after PV is deleted
+               
+               Milestone 4: Changes to the mounter interface as today it is assumed 'file' as the default.
+               
+               Milestone 5: Expose volumeType to users via kubectl
+               
+               Milestone 6: Adds enable/disable configuration to securityContext in PSP (Pod Security Policy) similar to hostPath
+               
+               Milestone 7: Validate container runtime options with user specifcations as indicated in UC3
+	       
+	       Milestone 8: Container Runtime changes
+
+Phase 2:  v1.9
+Feature: Discovery of block devices 
+
+                Milestone 1: Dynamically provisioned PVs to dynamically allocated devices
+
+                Milestone 2: Privileged container concerns    
+
+Other considerations:
+
+               Reference volume driver change(Attach/Detach logic) for pre-provisioned PVs
+
+               Reference volume driver change(Provision/Delete logic) for dynamic provisioning
+
+## Mounter interface proposed design
+
+```
+ type BlockMounter interface {
+	Mounter
+ 	CanSupportFileOnBlock() error
+            // kubelet needs volume's device path when attaches a volume to container
+ 	GetVolumePath() string
+ 	GetVolumeType() string   //TBD
+ }
+
+type BlockUnmounter interface {
+ 	Unmounter
+ 	GetVolumePath() string
+}
+```
+# Mounter binding matrix for statically provisioned volumes:
+
+| PV volumeType | PVC volumeType  | Result           |
+| --------------|:---------------:| ----------------:|
+|   unspecified | unspecified     | BIND             |
+|   file        | file            | BIND             |
+|   block       | unspecified     | NO BIND          |
+|   block       |  block          | BIND             |
+|   unspecified | block           | NO BIND          |
+|   block       |  file           | NO BIND          |
+|   file        |  block          | BIND**           |
+|  unspecified  | file            | BIND             |
+
+
+* unspecified defaults to 'file/ext4' today for backwards compatibility and in mount_linux.go  
+**this is plugin dependent 
+
+# Mounter binding matrix for dynamically provisioned volumes:
```

Comment:
The chapter should be probably called "dynamic provisioning matrix". `PV volumeType` column should be removed, PV is not existing at this time. And `Result` should be either `PV provisioned` (perhaps with some details how PV.volumeType / fsType is filled) or `error`.